2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,183 @@
+#Result-aggregation server for the installation-test scripts
+
+##Personal Details:
+
+* Name: Yashasvi Girdhar
+* Email: yash.girdhar@gmail.com
+* Personal Webpage : http://about.me/yashgirdhar
+* Github profile : https://github.com/itsyash
+* IRC nick (on freenode) : yash_
+* Telephone: +91-8143914414
+* Country of residence: India
+* Timezone: IST (GMT+5:30)
+* Primary language: English
+
+##Abstract:
+
+Software carpentry has installation scripts which are used by students to check if  they have correctly installed the required software for a workshop.
+
+The script basically performs a test for each software and outputs the result. In case of a failure, it prints the cause of failure and some diagnostic information on how to install the software correctly.
+Until now, we do not collect these results anywhere. But there are number of reasons why we should be doing that.
+
+First of all, it will help us figure out some important information, such as how much trouble the students are having installing some software or which packages need more attention.
+
+Some sample queries that the system would serve will look like:
+
+* how many tests failed in a particular workshop, on windows.
+* what fraction of failed tests were on OS x, for a specific time period.
+* for a particular workshop, find number of failed version tests for git.
+
+It will also help us to make decisions about using the right versions of the dependencies such as deprecating the older versions or adapting with the new versions.
+
+##Technical Details:
+
+The project is mainly divided into four phases:
+
+* Creating a model for storing the data in a relational database.
+
+The first step of the project is deciding what all information is useful to us and how we would be storing it. I have already built a first draft of the schema here, with feedback from the mentor.
```

Comment:
Also, I'd consider using an [ordered list][1] instead of bullet points that are followed by text like “The first step…” and “This step involves…”.  The information density of those extra words isn't very high, and tightening this sort of thing up will either make it a faster, punchier read or give you more space to go into project-specific details.

[1]: http://daringfireball.net/projects/markdown/syntax#list